utils/monitoring.js

```diff
@@ -0,0 +1,113 @@
+const { errors } = require('arsenal');
+const promClient = require('prom-client');
+const { Registry } = require('prom-client');
+const CountMaster = require('../CountItems/CountMaster');
+
+const aggregatorRegistry = new promClient.AggregatorRegistry();
+const { collectDefaultMetrics } = promClient;
+
+const bucketCount = (promClient.register.getSingleMetric('bucketCount_metrics'))
+    || new promClient.Gauge({
+        name: 'bucketCount_metrics',
+        help: 'Total number of buckets',
+        labelNames: ['bucketName'],
+    });
+
+// Histogram of the bucket processing duration, by the utilization service.
+
+const bucketProcessingDuration = (promClient.register.getSingleMetric('count_items_bucketProcessingDuration'))
+    || new promClient.Histogram({
+        name: 'count_items_bucketProcessingDuration',
+        help: 'Duration of processing a bucket',
+        labelNames: ['service'],
+        buckets: [1, 10, 60, 600, 3600, 18000, 36000],
+    });
+
+const consolidationDuration = (promClient.register.getSingleMetric('count_items_consolidationDuration'))
+    || new promClient.Histogram({
+        name: 'count_items_consolidationDuration',
+        help: 'Duration of metrics consolidation in seconds',
+        labelNames: ['service'],
+        buckets: [0.1, 0.5, 1, 2, 5, 10],
+    });
+
+const workersCount = (promClient.register.getSingleMetric('count_items_workersCountErrors'))
+    || new promClient.Counter({
+        name: 'count_items_workersCountErrors',
+        help: 'Number of errors in counting workers',
+        labelNames: ['state'],
+    });
+
+const objectsCount = (promClient.register.getSingleMetric('count_items_objectsCount'))
+    || new promClient.Counter({
+        name: 'count_items_objectsCount',
+        help: 'Number of objects',
+        labelNames: ['state'],
+    });
+
+const metricsCount = (promClient.register.getSingleMetric('count_items_metricsCount'))
+    || new promClient.Counter({
+        name: 'count_items_metricsCount',
+        help: 'Number of metrics',
+        labelNames: ['metricLevel'],
+    });
+/**
+ * @param {http.ServerResponse} res - http response object
+ * @param {errors.ArsenalError} error - Error code
+ * @return {void}
+ */
+function _writeResponse(res, error) {
+    let statusCode = 200;
+    if (error) {
+        if (Number.isInteger(error.code)) {
+            statusCode = error.code;
+        } else {
+            statusCode = 500;
+        }
+    }
```

Comment:
It seems to be always `MethodNotAllowed`, even in the `catch` case,, either we remove this logic, or we pass the real error from the exception?

Reply:
the real error is now returned , see here: https://github.com/scality/s3utils/pull/312/commits/5b7f15a750eb30d1dda2b2ed562563334cca9e57